src/System.Text.RegularExpressions/src/Properties/InternalsVisibleTo.cs

```diff
@@ -0,0 +1,7 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+
```

Comment:
We've tried to avoid doing this,  preferring reflection if necessary instead. Other issues aside, now that this will prevent the Linker from shaking away dead code.
cc: @weshaggard

Reply:
I wouldn't be able to use attribute based testing if the `RegexParseError` enum isn't public: https://github.com/dotnet/corefx/pull/29178/files#diff-e55ccc955dc3ddac57a0e77d1a8fc944R947 and I wouldn't be able to catch on a the specific exception. I could workaround the second issue easily by catching the parent exception and then checking the exact type via reflection and its error code but I'm not sure how to handle the first issue. We are speaking about ~1000 test cases here, switching to strings or ints instead of the enum value doesn't seem right?

Reply:
Yes lets please try and avoid InternalsVisibleTo as it causes coupling that is too tight to one particular implementation.

Reply:
Using reflection instead of IVT also makes it easier for us in Mono to consume the tests since our test assemblies have different names 😄

Reply:
Great, thanks for the input.